src/services/utilities.ts

```diff
@@ -857,6 +857,15 @@ namespace ts {
         return false;
     }
 
+    export function isBreakOrContinue(kind: SyntaxKind) {
+        return kind === SyntaxKind.BreakKeyword || kind === SyntaxKind.ContinueKeyword;
+    }
+
+    export function isInBreakOrContinue(sourceFile: SourceFile, position: number): boolean {
+        const contextToken = findPrecedingToken(position, sourceFile);
+        return contextToken && (isBreakOrContinue(contextToken.kind) || contextToken.parent && isIdentifier(contextToken) && isBreakOrContinueStatement(contextToken.parent));

```

Comment:
we already do this in getCompletionData, consider moving the check for isLabeledCompletion to that function.